### ISSUE TITLE
[Proposal] BindZip to store both income and outcome of Bind

### ISSUE THREAD

**Issue by hankovich** [vkhorikov/CSharpFunctionalExtensions#535]
I have a lot of code like

```csharp
GetEntity(entityId)
    .Bind(entity => GetNestedEntity(entity.NestedEntityId).Map(nestedEntities => (entity, nestedEntities));

Result<Entity> GetEntity(EntityId id) => throw null!;

Result<NestedEntity> GetNestedEntity(NestedEntityId id) => throw null!;
```

i.e. in further processing I need both income and outcome of `Bind`. I suggest to add a new method `BindZip`

```c#
GetEntity(entityId)
    .BindZip(entity => GetNestedEntities(entityId)); // returns Result<(Entity First, NestedEntity Second)>
```

**Comment by hankovich** [comment]
It's not only me who needs such overload: https://github.com/vkhorikov/CSharpFunctionalExtensions/discussions/534

**Comment by vkhorikov** [comment]
Sorry for the late reply.
Sure, let's add this.

**Comment by DerStimmler** [comment]
I think this issue can be closed since #540 has been merged?
